### PR TITLE
SDC-3435

### PR DESCRIPTION
**Pull request by aharonha**
Ability to add file name suffixes

### CODE REVIEW

**Comment by aharonha** [comment]
This a squash rebased PR

**Comment by madhukard** [comment]
@aharonha : In order to become a contributor to StreamSets open-source projects on Github you must sign the Contributor License Agreement.   - https://streamsets.com/contributing/

File: aws-lib/src/main/java/com/streamsets/pipeline/stage/destination/s3/S3TargetConfigBean.java [hunk 1]
```diff
@@ -74,6 +74,17 @@
     group = "S3"
   )
   public String fileNamePrefix;
+  
+  @ConfigDef(
+    required = true,
+    type = ConfigDef.Type.STRING,
+    defaultValue = "",
+    description = "Suffix for file names that will be uploaded on Amazon S3, No need for .gz when compression is selected",
```
Comment (jarcec) [inline aws-lib/src/main/java/com/streamsets/pipeline/stage/destination/s3/S3TargetConfigBean.java]: Can we put an example of expected suffix here. Something like "Suffix for file names that will be .. (for example '.txt.') ..."

File: hdfs-protolib/src/main/java/com/streamsets/pipeline/stage/destination/hdfs/HdfsTargetConfigBean.java [hunk 2]
```diff
@@ -141,7 +141,19 @@
...
   )
   public String uniquePrefix;
-
+  
+  @ConfigDef(
+    required = false,
+    type = ConfigDef.Type.STRING,
+    defaultValue = "",
+    label = "Files Suffix",
+    description = "File name suffix",
```
Comment (jarcec) [inline hdfs-protolib/src/main/java/com/streamsets/pipeline/stage/destination/hdfs/HdfsTargetConfigBean.java]: Let's use nice and long description similarly as we have in the S3 destination.

File: hdfs-protolib/src/main/java/com/streamsets/pipeline/stage/destination/hdfs/HdfsTargetConfigBean.java [hunk 3]
```diff
@@ -528,7 +543,8 @@ public void init(Stage.Context context, List<Stage.ConfigIssue> issues) {
                   new URI(hdfsUri),
                   hdfsConfiguration,
                   uniquePrefix,
-                  false, // Late records doesn't support "template directory" to be in header
+                  fileNameSuffix, // Late records doesn't support "template directory" to be in header
```
Comment (jarcec) [inline hdfs-protolib/src/main/java/com/streamsets/pipeline/stage/destination/hdfs/HdfsTargetConfigBean.java]: Can we perhaps not do this unrelated change?

File: hdfs-protolib/src/main/java/com/streamsets/pipeline/stage/destination/hdfs/HdfsTargetConfigBean.java [hunk 4]
```diff
@@ -540,8 +556,7 @@ public void init(Stage.Context context, List<Stage.ConfigIssue> issues) {
                   keyEl,
                   false,
                   null,
-                  dataGeneratorFormatConfig.getDataGeneratorFactory(),
-                  (Target.Context) context, "lateRecordsDirPathTemplate"
+                  dataGeneratorFormatConfig.getDataGeneratorFactory(), (Target.Context) context, "lateRecordsDirPathTemplate"
```
Comment (jarcec) [inline hdfs-protolib/src/main/java/com/streamsets/pipeline/stage/destination/hdfs/HdfsTargetConfigBean.java]: Can we perhaps not do this unrelated change?

File: hdfs-protolib/src/test/java/com/streamsets/pipeline/stage/destination/hdfs/writer/RecordWriterManagerTestBuilder.java [hunk 5]
```diff
@@ -40,6 +40,7 @@
   private URI hdfsUri = new URI("file:///");
   private Configuration hdfsConf = new HdfsConfiguration();
   private String uniquePrefix = "prefix";
+  private String fileNameSuffix="";
```
Comment (jarcec) [inline hdfs-protolib/src/test/java/com/streamsets/pipeline/stage/destination/hdfs/writer/RecordWriterManagerTestBuilder.java]: Can we expose this in the builder?

**Comment by jarcec** [comment]
We should also include additional tests for HDFS Destination.

**Comment by jarcec** [comment]
Wondering if you've seen my comments @aharonha ?

**Comment by aharonha** [comment]
I can not sign the CLA yet, our legal department are checking it.

**Comment by jarcec** [comment]
Let me know if there is anything that we can do to help signing the CLA @aharonha .

In the mean time, I'll need to ask you to rebase your patch on current top of the current branch as we needed to do a force push.

**Comment by jarcec** [comment]
Wondering if you have any luck with getting the approval to sign the CLA @aharonha ? I would love to get this in as multiple users asked for the functionality :)

We've also simplified the process a bit in the mean time, so I would encourage you to check the following page with updated instructions:

https://streamsets.com/contributing/

**Comment by utenakr** [comment]
Hi @aharonha,
Just wondering if you are still interested in submitting this patch? There has been a lot of changes since your last commit. You will need to rebase the latest master. Let me know if you need any assistance.

**Comment by utenakr** [comment]
@aharonha, my understanding is that you've been waiting for the approval from your legal department to sign the CLA back in Jul 28, 2016. We have a high demand on this feature from other customers, so we decide to move on. Again, thank you for your contribution and let me know if you have any concerns.
